### PR TITLE
Actions to replace Travis' Coverage reporting (removes Travis)

### PR DESCRIPTION
**Pull request by LaurenceBeard**
- Removes Travis
- Adds cov-report to pytest run and task to post to codacy/codecov
- Splits style checking into another job for speed

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-  # Install our own extra dependencies (+ filelock for Iris test).
-  - conda install -c conda-forge astroid=2.1.0 filelock mock netcdf4=1.4.1 numpy=1.15.4 pycodestyle=2.3.1 pylint=2.1.1 pandas=0.23.4 python-stratify=0.1 sphinx=1.8.1 coverage pytest pytest-cov
-  - pip install codacy-coverage codecov clize sigtools
-
-  # List the name and version of all the dependencies within the conda environment.
-  - conda list
-
-script:
-  - python -c "import iris"
-  - bin/improver-tests
```
[the remark this answers is not in the document]
Comment (TomekTrzeciak) [reply .travis.yml]: Do we even still need `bin/improver-tests`?

[the remark this answers is not in the document]
Comment (TomekTrzeciak) [reply .travis.yml]: Asked on teams.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -3,30 +3,47 @@ name: Tests
 on: [push, pull_request]
```
Comment (TomekTrzeciak) [inline .github/workflows/tests.yml]:
```suggestion
on: [pull_request]
```

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomekTrzeciak) [reply .travis.yml]: And the answer is yes.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -12,21 +11,43 @@ jobs:
       with:
         path: /usr/share/miniconda/envs/improver
```
[the remark this answers is not in the document]
Comment (LaurenceBeard) [reply .github/workflows/tests.yml]: I agree, it's not the nicest, but I'm not sure the matrix build makes it much neater.
It it's hard to run the cov-upload in the same job as the pytest step.
You could run as another job, but then you'd have to use artifacts to upload/download the report, a bit of a faff and makes it more complicated.
Feel free to have a go at refactoring, but GH Actions don't make such things easy.

[the remark this answers is not in the document]
Comment (TomekTrzeciak) [reply .github/workflows/tests.yml]: GH should have made actions work like [this](https://circleci.com/docs/2.0/migrating-from-github):
> CircleCI orbs are just packaged, reusable YAML, so you can orbify reusable jobs, executors, or commands, and use them however you see fit in any of your jobs or workflows.

[the remark this answers is not in the document]
Comment (tjtg) [reply .github/workflows/tests.yml]: Ok, sounds like the current solution is the best that can be done in the situation.

**Pull request merged**